python/cudf/cudf/core/dataframe.py

```diff
@@ -843,8 +843,13 @@ def _init_from_list_like(self, data, index=None, columns=None):
             data = DataFrame.from_pandas(pd.DataFrame(data))
             self._data = data._data
         else:
-            data = list(itertools.zip_longest(*data))
-
+            if all(
+                isinstance(col, (abc.Iterable, abc.Sequence))
+                for col in data
+            ):
+                data = list(itertools.zip_longest(*data))
+            else:
+                raise TypeError("Inputs should be an iterable or sequence.")
```

Comment:
Nit: maybe write “if not… raise” and then there’s no else required. Control flow can continue with “data = …”

Reply:
Yup, updated 👍